### PR TITLE
allow passing database id into notion reader

### PR DESCRIPTION
**Pull request by bborn**
If a `database_id` is passed, just index all the pages in the database.

### CODE REVIEW

File: gpt_index/readers/notion.py [hunk 1]
```diff
@@ -117,7 +133,7 @@ def search(self, query: str) -> List[str]:
                 next_cursor = data["next_cursor"]
         return page_ids
 
-    def load_data(self, page_ids: List[str]) -> List[Document]:
+    def load_data(self, page_ids: List[str] = [], database_id="") -> List[Document]:
```
Comment (jerryjliu) [inline gpt_index/readers/notion.py:136]: i'd type this as Optional[None]

File: examples/data_connectors/NotionDemo.ipynb [hunk 2]
```diff
@@ -11,7 +11,7 @@
   },
```
Comment (jerryjliu) [inline examples/data_connectors/NotionDemo.ipynb:11]: the jupyter notebook seems to run into issues? does it run?

**Review by jerryjliu — COMMENTED** [review]
thanks! a few comments

File: examples/data_connectors/NotionDemo.ipynb (hunk 2 above)
Comment (bborn) [reply examples/data_connectors/NotionDemo.ipynb:11]: It was running fine for me? What did you give it for database_id param?

File: gpt_index/readers/notion.py [hunk 3]
```diff
@@ -93,6 +93,22 @@ def read_page(self, page_id: str) -> str:
         """Read a page."""
         return self._read_block(page_id)
 
+    def query_database(self, database_id: str, query_dict: Dict[str, Any] = {}) -> List[str]:
+        """Get all the pages from a Notion database."""
+        
+        database_url = f"https://api.notion.com/v1/databases/{database_id}/query"
```
Comment (jerryjliu) [inline gpt_index/readers/notion.py:99]: could you move this to a global var at top of file? (consistent with other global vars)

**Comment by bborn** [comment]
Nice! You beat me to it! Sorry - traveling today so didn't have time to fix this up.

**Pull request merged**